### PR TITLE
Throw error if none of the db rows are affected

### LINKED ISSUE #181: Change the behavior of update() to return error when rows_affected is 0
`update()` can be used with filters. One might, for example, update a record but only if some column of itself still has some value set.

example:

```rust
MyEntity::update(ActiveRecord {
    id: Set(5),
    status: Set("ACTIVE"),
})
.filter(column::Status.eq("READY_FOR_ACTIVATION")
.exec(&db)
.await
```

This is useful to handle idempotency problems: I want to update status but only if, in the meantime, it hasn't yet been updated by another oprocess or something similar.

In this situation (or if I provide an id which does not exist) I have to know if my update has occurred successfully.

One can arg that I could use update_many, but using update gives me the advantage to have the active model itself be returned, which as far as I understand is not the case with update_many.

_Originally posted by @marlon-sousa in https://github.com/SeaQL/sea-orm/discussions/127#discussioncomment-1358559_

We should add a new Error variant, perhaps `RecordNotFound`

### PR DESCRIPTION
**Pull request by billy1624**
Resolve #181

### CODE REVIEW

**Comment by tyt2y3** [comment]
We need unit tests using the mock connection on this behaviour too

**Comment by billy1624** [comment]
> We need unit tests using the mock connection on this behaviour too

Added!

File: src/executor/update.rs [hunk 1]
```diff
@@ -73,7 +73,117 @@ where
...
+
+#[cfg(test)]
+mod tests {
+    use crate::{entity::prelude::*, tests_cfg::*, *};
+    use pretty_assertions::assert_eq;
+
+    #[smol_potat::test]
+    async fn update_record_not_found_1() -> Result<(), DbErr> {
+        let db = MockDatabase::new(DbBackend::Postgres)
+            .append_query_results(vec![
```
Comment (tyt2y3) [inline src/executor/update.rs:94]: I guess this is not needed. Let me try.

**Review by tyt2y3 — CHANGES_REQUESTED** [review]
I have noticed a problem. this behavior should only applies to update_one, while for update many, it's fine to have updated 0 rows.

**Comment by billy1624** [comment]
> I have noticed a problem. this behavior should only applies to update_one, while for update many, it's fine to have updated 0 rows.

Fixed

**Review by tyt2y3 — APPROVED** [review]

**Pull request merged**